test/all_tests.cpp

```diff
@@ -546,6 +546,22 @@ TEST(OptionTest, destruct_after_take) {
   EXPECT_TRUE(did_destruct);
 }
 
+//------------------------------------------------------------------------------
+TEST(OptionTest, empty_base) {
+  struct Empty {
+    bool operator==(const Empty&) const { return true; }
+  };
+  auto op1 = fun::some(Empty());
+  EXPECT_TRUE(op1.is_some());
+  fun::Option<Empty> op2 {};
+  EXPECT_TRUE(op2.is_none());
+  op2 = std::move(op1);
+  EXPECT_TRUE(op2.is_some());
+  EXPECT_TRUE(op1.is_none());
+  op1.emplace(Empty());
+  EXPECT_TRUE(op1 == op2);
+}
```

Comment:
Please use `fun::Unit` instead of `Empty`. The following copy should be an easy copy-and-paste fix.

```
TEST(OptionTest, empty_base) {
  auto op1 = fun::some(fun::Unit());
  EXPECT_TRUE(op1.is_some());
  fun::Option<fun::Unit> op2 {};
  EXPECT_TRUE(op2.is_none());
  op2 = std::move(op1);
  EXPECT_TRUE(op2.is_some());
  EXPECT_TRUE(op1.is_none());
  op1.emplace(fun::Unit());
  EXPECT_TRUE(op1 == op2);
}
```